File: C++/src/CuentaCifraApp.cpp
#include <iostream>
#include <string>

using namespace std;

int main(int argc, char const *argv[])
{
    cout << "deme un numero \n";
    int number;

    cin >> number;

    if (number < 0)
    {
        cout << "el numero es negativo\n";
    }
    else
    {
        string Cadena = std::to_string(number); // convierte el numero a string

        cout << "el numero ingresado tiene " << Cadena << " cifras\n";
    }

    return 0;
}

File: java/src/CuentaCifraApp.java
import java.util.Scanner;

public class CuentaCifraApp {
    public static void main(String[] args) {
        // Pide por teclado un número entero positivo (debemos controlarlo) y muestra el
        // número de cifras que tiene. Por ejemplo: si introducimos 1250, nos muestre
        // que tiene 4 cifras. Tendremos que controlar si tiene una o mas cifras, al
        // mostrar el mensaje.
        Scanner input = new Scanner(System.in);

        System.out.println("deme un numero");
        int entero = input.nextInt();

        if (entero < 0) {
            System.out.println("el numero es negativo");
            System.exit(-1);
        }

        String Cadena = Integer.toString(entero);// este convierte el int a String

        System.out.println("el numero ingresado tiene " + Cadena.length() + " cifras");

    }
}
